server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
We have intentionally implemented this in a fully backwards compatible manner that maintains the old behavior while allowing us to start figuring out what is needed to make things better.  

So, the purpose of the PR is to get something that works better in a limited use case that we specifically have.  And, in the process, move things in a good forward direction.  Given that this doesn't actually change old behavior and I think it's been doing a good job of informing and concretizing the future changes that are going to be required to really harden this.